src/parser/regexp.bnf

```diff
@@ -103,7 +103,14 @@ GROUP_NAME              ([\w$]|\\'u'[0-9a-fA-F]{4}|\\'u{'[0-9a-fA-F]{1,}'}')+
 
 {ESC}'/'                            return 'ESC_CHAR'
 
-{ESC}{CHAR}                         return 'ESC_CHAR'
+{ESC}{CHAR}                         {
+                                        const s = this.getCurrentState();
+                                        if (s === 'u' || s === 'xu' || s === 'u_class') {
+                                            throw new SyntaxError(`invalid unicode escape ${yytext}`);
```

Comment:
Nit: `s/unicode/Unicode/`

Reply:
If this is only about `\p{…}`, how about: “invalid Unicode property escape”?

Reply:
This is not only about `\p{...}`. All redondant escapes (`\e`, `\a`, `\g`...) will throw this error when `u` flag is set.